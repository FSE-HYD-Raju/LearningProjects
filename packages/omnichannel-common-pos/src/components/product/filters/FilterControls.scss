@import '../../../variables';
@import '../../../styles/mixins';

.b2c-products-filter{

	&-main {
		display: flex;
		background: $light-grey-lighter;
		margin: 1rem 0;
		background: #f8f8f8;
		padding: 0 0.5rem;
	}

	&-price-range {
		display: flex;
		flex: 1;
		flex-direction: column;
		margin-top: 0;
		min-width: 300px;
		max-width: 300px;
		width: 300px;

		&-option {
			max-height: $form-field-line-height;
			margin-bottom: 0;
			&.custom-radio {
				margin-bottom: 0.625rem;
			}
		}

		&-slider {
			margin-top: 10px;
			display: flex;
			flex: 1;
			align-items: center;
			justify-content: space-between;

			&-container {
				display: flex;
				flex-direction: column;
				margin-left: 6px;
				margin-top: 0.625rem;
			}
			&-legit {
				opacity: 1.0;
				pointer-events: auto;
			}
			&-nonlegit {
				opacity: 0.35;
				pointer-events: none;
			}
			&-label {
				font-size: $form-field-text-font-size;
				color: $graphite-lighter;
				&-currency {
					font-weight: bold;
				}
			}
		}
	}

	&-inner-container {
		display: flex;
		flex-direction: column;
	}

	&-minutes-range {
		min-width: 300px;
		margin-left: 12px;

	}

	&-container {
		padding: 4px;
		margin-right: 60px;
	}

	&-title {
		@include form-field-text;
		padding-bottom: 0.5rem;
	}

	&-voice {
		max-height: $form-field-line-height;
		margin-bottom: 20px;
	}
}
