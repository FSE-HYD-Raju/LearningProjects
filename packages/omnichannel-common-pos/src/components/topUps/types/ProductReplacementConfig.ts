interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
