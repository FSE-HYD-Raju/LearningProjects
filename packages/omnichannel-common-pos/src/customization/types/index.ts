import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
