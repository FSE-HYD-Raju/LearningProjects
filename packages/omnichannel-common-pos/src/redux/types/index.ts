import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
