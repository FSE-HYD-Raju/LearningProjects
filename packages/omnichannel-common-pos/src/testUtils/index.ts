import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
