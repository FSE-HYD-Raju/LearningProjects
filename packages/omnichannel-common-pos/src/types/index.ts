import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
