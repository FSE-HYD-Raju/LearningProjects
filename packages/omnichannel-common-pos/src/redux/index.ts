"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
