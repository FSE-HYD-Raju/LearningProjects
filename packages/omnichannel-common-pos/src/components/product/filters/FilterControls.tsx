import * as React from "react";
import OcInput from "../../ocComponents/OcInput";
import { get, debounce } from "lodash";
import CheckboxCombinedFilter from "../../filters/CheckboxCombinedFilter";
import CheckboxListProductFilter from "../../filters/CheckboxListProductFilter";
import classnames from "classnames";
import { ProductFilter, ProductFilterRangeType, DisplayTypeEnum, ProductFilterTypeEnum } from "../../../redux/types";
import { ProductPriceEnum } from "../ProductPriceType";
import messages from "./FilterControls.messages";
import { FormattedMessage, FormattedNumber } from "../../../channelUtils";
import { FilterControlsStateProps, FilterControlsActionProps } from "./FilterControlsContainer";
// This ugly magic is here due to wrong typings for react-input-range, sorry for eyes bleeding
import { default as InputRangeComponent, Range } from "react-input-range";

const InputRange: typeof InputRangeComponent = require("react-input-range");

type FilterControlsProps = FilterControlsStateProps & FilterControlsActionProps;

interface FilterControlsState {
	selectedRange: Record<string, Range>;
}

class FilterControls extends React.PureComponent<FilterControlsProps, FilterControlsState> {

	private updateFilterDebounced = debounce<(value: any) => void>((value: any) => {
			this.props.actions.handlePriceRangeSlider(value);
		},
		100,
		{ trailing: true, leading: false }
	);

	private upateMinutesFilterDebounced = debounce<(value: any) => void>(
		value => {
			this.props.actions.updateMinutesFilterValues(value);
		},
		100,
		{ trailing: true, leading: false }
	);

	constructor(props: FilterControlsProps) {
		super(props);
		this.state = {
			selectedRange: {},
		};
	}

	rangeSliderClassNames = () => {
		return {
			activeTrack: "RangeSlider-input-range__track--active",
			disabledInputRange: "RangeSlider-input-range--disabled",
			inputRange: "RangeSlider-input-range",
			labelContainer: "RangeSlider-input-range__slider-container",
			maxLabel: "RangeSlider-input-range__label--max",
			minLabel: "RangeSlider-input-range__label--min",
			slider: "RangeSlider-input-range__slider",
			sliderContainer: "input-range__slider-container",
			track: "RangeSlider-input-range__track",
			valueLabel: "RangeSlider-input-range__label--valueLabel"
		};
	};

	componentWillMount() {
		this.setState({
			selectedRange: {...this.props.priceRange}
		});
	}

	componentWillReceiveProps(newprops: FilterControlsProps) {
		if (this.props.products !== newprops.products) {
			this.props.actions.updatePriceRange(newprops.products, newprops.onlyFilterUpdated);
		}

		if (!newprops.onlyFilterUpdated) {
			this.setState({
				selectedRange: {...newprops.priceRange}
			});
		}
	}

	handlePriceRangeSlider = (value: any) => {
		const key = this.props.priceType;
		this.setState({
			selectedRange: {
				...this.state.selectedRange,
				[key!]: value
			}
		});

		this.updateFilterDebounced(value);
	};

	handleMinutesRangeSlider = (minutesRange: any) => {
		this.upateMinutesFilterDebounced(minutesRange);
	};

	toggleApplyIncludedMinutesFilter = () => {
		this.props.actions.toggleApplyIncludedMinutesFilter();
	};

	toggleFilter = (e: React.SyntheticEvent<any>, id: string, key: string) => {
		e.preventDefault();
		this.props.actions.toggleFilter({id, filterKey: key});
	};

	renderPriceRangeFilter = (priceRange: any) => {
		const hasUsageRange = false;

		let priceRangeIsLegit = false;
		if (priceRange) {
			priceRangeIsLegit = priceRange.min < priceRange.max;
		}

		const boundariesForPriceRange = this.props.actions.getBoundariesForPriceRange(this.props.products);
		const hasAllPricesRange = boundariesForPriceRange && boundariesForPriceRange.allPrices;
		const hasUpfrontRange = boundariesForPriceRange && boundariesForPriceRange.upfront;
		const hasRecurringRange = boundariesForPriceRange && boundariesForPriceRange.recurring;

		return (
			<div className="b2c-products-filter-price-range">
				<div className="b2c-products-filter-inner-container">
					<OcInput
						label={<FormattedMessage {...messages.allCosts}/>}
						type="radio"
						className="b2c-products-filter-price-range-option"
						checked={this.props.priceType === ProductPriceEnum.ALL_PRICES}
						onChange={() => this.props.actions.switchPriceType(ProductPriceEnum.ALL_PRICES)}
						id="shop-container-filters-all-costs"
					/>
					{hasUpfrontRange && (
						<OcInput
							label={<FormattedMessage {...messages.oneTimeCost}/>}
							type="radio"
							className="b2c-products-filter-price-range-option"
							checked={this.props.priceType === ProductPriceEnum.UPFRONT}
							onChange={() => this.props.actions.switchPriceType(ProductPriceEnum.UPFRONT)}
							id="shop-container-filters-one-time-cost"
						/>
					)}
					{hasRecurringRange && (
						<OcInput
							label={<FormattedMessage {...messages.recurringCost}/>}
							type="radio"
							className="b2c-products-filter-price-range-option"
							checked={this.props.priceType === ProductPriceEnum.RECURRING}
							onChange={() => this.props.actions.switchPriceType(ProductPriceEnum.RECURRING)}
							id="shop-container-filters-recurring-cost"
						/>
					)}
					{hasUsageRange && (
						<OcInput
							label={<FormattedMessage {...messages.usageCost}/>}
							type="radio"
							className="b2c-products-filter-price-range-option"
							checked={this.props.priceType === ProductPriceEnum.USAGE}
							onChange={() => this.props.actions.switchPriceType(ProductPriceEnum.USAGE)}
							id="shop-container-filters-usage-costs"
						/>
					)}
				</div>

				{(hasAllPricesRange || hasUpfrontRange || hasRecurringRange || hasUsageRange) && (
					<div className="b2c-products-filter-price-range-slider-container">
						<InputRange
							maxValue={priceRange.max}
							minValue={priceRange.min}
							value={this.props.priceType ? this.state.selectedRange[this.props.priceType] : 0}
							onChange={this.handlePriceRangeSlider}
							step={1}
							disabled={!priceRangeIsLegit}
							classNames={this.rangeSliderClassNames()}
						/>
						<div
							className={
								classnames("b2c-products-filter-price-range-slider",
								priceRangeIsLegit
									? "b2c-products-filter-price-range-slider-legit"
									: "b2c-products-filter-price-range-slider-nonlegit"
							)}
						>
							<span className="b2c-products-filter-slider-label">
								<FormattedMessage {...messages.from} />
								{" "}
								<span className="b2c-products-filter-slider-label-currency">
									<FormattedNumber
										style="currency"
										currency="EUR"
										value={this.props.priceType && this.state.selectedRange[this.props.priceType] ?
											this.state.selectedRange[this.props.priceType].min : 0}
										minimumFractionDigits={0}
										maximumFractionDigits={0}
									/>
								</span>
							</span>

							<span className="b2c-products-filter-slider-label">
								<FormattedMessage {...messages.to}/>
								{" "}
								<span className="b2c-products-filter-slider-label-currency">
									<FormattedNumber
										style="currency"
										currency="EUR"
										value={this.props.priceType && this.state.selectedRange[this.props.priceType] ?
											this.state.selectedRange[this.props.priceType].max : 0}
										minimumFractionDigits={0}
										maximumFractionDigits={0}
									/>
								</span>
							</span>
						</div>
					</div>
				)}
			</div>
		);
	};

	renderMinutesRangeFilter = () => {
		const { minutesRange, minutesFilterValues } = this.props;
		const minValue = minutesRange!.min;
		const maxValue = minutesRange!.max;
		const currentValues: ProductFilterRangeType = {
			min: get(minutesFilterValues, "min") || minValue,
			max: get(minutesFilterValues, "max") || maxValue
		};

		const currentMin = currentValues.min;
		const currentMax = currentValues.max;

		return (
			<div className="b2c-products-filter-minutes-range">
				<InputRange
					maxValue={maxValue}
					minValue={minValue}
					value={currentValues}
					onChange={this.handleMinutesRangeSlider}
					step={100}
					disabled={!this.props.applyIncludedMinutesFilter}
					classNames={this.rangeSliderClassNames()}
				/>
				<div
					className={classnames(
						"b2c-products-filter-price-range-slider",
						"b2c-products-filter-price-range-slider-legit"
					)}
				>
					<span className="b2c-products-filter-slider-label">
						<FormattedMessage {...messages.fromMin} />
						<span className="b2c-products-filter-slider-label-currency">
							<span>{currentMin}</span>
							<FormattedMessage {...messages.min} />
						</span>
					</span>
					<span className="b2c-products-filter-slider-label">
						<FormattedMessage {...messages.inputRangeMaxLabel} />
						<span className="b2c-products-filter-slider-label-currency">
							<span>{currentMax}</span>
							<FormattedMessage {...messages.toMin} />
						</span>
					</span>
				</div>
			</div>
		);
	};

	renderFilterControls = (filter: ProductFilter, priceRange: ProductFilterRangeType | undefined, activeFilters: Array<ProductFilter>): React.ReactNode => {
		if (!priceRange && !filter) {
			return null;
		}
		// RND-12950-vesak: let's hide usage for now until we know how to show usage prices
		// const hasUsageRange = this.props.getBoundariesForPriceRange(
		// 	(this.props.products: Array<ProductOffering | ProductOfferingGroup>)
		// )["usage"];

		return (
			<div
				key={filter.type + filter.id}
				className="b2c-products-filter-container"
			>
				<span className="b2c-products-filter-title">
					{(filter.filterLabel === "price-range" && (
						<span>
							<FormattedMessage {...messages.priceRange}/>
						</span>
					)) ||
						(filter.filterLabel === "voice" && (
							<div>
								<OcInput
									className="b2c-products-filter-voice"
									label={
										<span className="b2c-products-filter-title">
											<FormattedMessage {...messages.includedMinutes}/>
										</span>
									}
									type="checkbox"
									checked={this.props.applyIncludedMinutesFilter}
									onChange={this.toggleApplyIncludedMinutesFilter}
									id="shop-container-filters-apply-included-minutes"
								/>
							</div>
						))}
				</span>
				{filter.type === "price-range" && priceRange && this.renderPriceRangeFilter(priceRange)}
				{filter.filterLabel === "voice" && this.renderMinutesRangeFilter()}

				{filter.displayType === DisplayTypeEnum.DROPDOWN && (
					<CheckboxListProductFilter
						toggleFilter={this.toggleFilter}
						filter={filter}
						activeFilters={activeFilters}
					/>
				)}
				{filter.displayType === DisplayTypeEnum.COMBINED_CHECKBOX && (
					<CheckboxCombinedFilter
						toggleCombinedFilter={this.props.actions.toggleCombinedFilter}
						filter={filter}
						combinedFilters={this.props.combinedFilters}
					/>
				)}
			</div>
		);
	};

	render() {
		const { filters, products, activeFilters, priceType } = this.props;
		const priceRange = this.props.actions.getBoundariesForPriceRange(products);
		const rangeForType: ProductFilterRangeType | undefined = priceType && priceRange && priceRange[priceType];

		const priceFilter = filters && filters.find(filter => filter.type === ProductFilterTypeEnum.PRICE_RANGE);

		return (
			<div>
				<div className="b2c-products-filter-main">
					{priceFilter && rangeForType && this.renderFilterControls(priceFilter, rangeForType, activeFilters)}
					{filters && filters.length > 0 && filters.filter(filter => filter.type !== ProductFilterTypeEnum.PRICE_RANGE)
							.map(filter => {
								return this.renderFilterControls(filter, rangeForType, activeFilters);
							})}
				</div>
			</div>
		);
	}
}

export default FilterControls;
export {
	FilterControlsProps,
	FilterControlsState,
};
