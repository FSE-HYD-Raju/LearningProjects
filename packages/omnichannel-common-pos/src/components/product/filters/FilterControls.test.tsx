import * as React from "react";
import FilterControls, { FilterControlsProps } from "./FilterControls";
import { shallowWithContext } from "../../../testUtils";
import { ProductFilter, ProductOffering } from "../../../redux/types";

// TODO: update tests to test at least something
describe("FilterControls", () => {
	const minProps: FilterControlsProps = {
		products: [] as any as Array<ProductOffering>,
		activeFilters: [] as any as Array<ProductFilter>,
		filters: [] as any as Array<ProductFilter>,
		showFilterControls: true,
		applyIncludedMinutesFilter: false,
		combinedFilters: {},
		onlyFilterUpdated: false,
		actions: {
			getBoundariesForPriceRange: jest.fn(),
			updatePriceRange: jest.fn(),
			switchPriceType: jest.fn(),
			handlePriceRangeSlider: jest.fn(),
			updateMinutesFilterValues: jest.fn(),
			toggleFilter: jest.fn(),
			toggleCombinedFilter: jest.fn(),
			toggleApplyIncludedMinutesFilter: jest.fn(),
		}
	} as any as FilterControlsProps;

	it("succeeds at shallow mount without props", () => {
		const wrapper = shallowWithContext(<FilterControls {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
