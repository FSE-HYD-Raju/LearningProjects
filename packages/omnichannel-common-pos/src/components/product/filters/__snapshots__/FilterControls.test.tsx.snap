// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterControls succeeds at shallow mount without props 1`] = `
<div>
  <div
    className="b2c-products-filter-main"
  />
</div>
`;
